kolibri/utils/server.py

```diff
@@ -334,6 +334,19 @@ def run_server(port):
     # Subscribe this server
     server.subscribe()
 
+    # Modify the default signal handler logic for cherrypy to make it also execute
+    # the previously subscribed signal handler.
+    original_handle_signal = cherrypy.engine.signal_handler._handle_signal
+
+    def _handle_signal(self, signum=None, frame=None):
+        original_handle_signal(self, signum=signum, frame=frame)
+        if callable(self._previous_handlers[signum]):
```

Comment:
Never going to be the case that this is a `KeyError`? Cherrypy sets defaults, or it's a defaultdict or something?